File: PyScripts/Walmart Example/summarize_output.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Jun 24 16:41:02 2021

@author: clark
"""
import numpy as np
def summarize_fitp(fitp):
     print(fitp['mdraws'][0, 0:5]); print(np.mean(fitp['mdraws']))
     print(fitp['sdraws'][0, 0:5]); print(np.mean(fitp['sdraws']))
     # print(fitp['mmean'][0:5]); print(np.mean(fitp['mmean']))
     # print(fitp['smean'][0:5]); print(np.mean(fitp['smean']))
     print(fitp['msd'][0:5]); print(np.mean(fitp['msd']))
     print(fitp['ssd'][0:5]); print(np.mean(fitp['ssd']))
     print(np.mean(fitp['m_5'])); print(np.mean(fitp['m_lower'])); print(np.mean(fitp['m_upper']))
     print(np.mean(fitp['s_5'])); print(np.mean(fitp['s_lower'])); print(np.mean(fitp['s_upper']))
def summarize_fitv(fitv):
     print(fitv['vdraws'][0:3, :]); print(np.mean(fitv['vdraws']))
     print(fitv['vdrawsh'][0:3, :]); print(np.mean(fitv['vdrawsh']))
     print(fitv['mvdraws']); print(fitv['mvdrawsh'])
     print(fitv['vdraws_sd']); print(fitv['vdrawsh_sd']); print(fitv['vdraws_5'])
     print(fitv['vdrawsh_5']); print(fitv['vdraws_lower']); print(fitv['vdraws_upper'])
     print(fitv['vdrawsh_lower']); print(fitv['vdrawsh_upper'])
def summarize_fits(fits):
     print(np.mean(fits['vidraws'])); print(np.mean(fits['vijdraws']))
     print(np.mean(fits['tvidraws'])); print(np.mean(fits['vdraws']))
     print(np.mean(fits['sidraws'])); print(np.mean(fits['sijdraws']))
     print(np.mean(fits['tsidraws']))
     # print(fits['msi']); print(fits['msi_sd']); print(fits['si_5'])
     # print(fits['si_lower']); print(fits['si_upper']); print(fits['msij'])
     # print(fits['sij_sd']); print(fits['sij_5']); print(fits['sij_lower'])
     # print(fits['sij_upper']); print(fits['mtsi']); print(fits['tsi_sd'])
     # print(fits['tsi_5']); print(fits['tsi_lower']); print(fits['tsi_upper'])
def save_fits(fits, fname):
     from pathlib import Path
     to_save = dict(fits) # A "shallow copy"
     for cat in ('vidraws', 'vijdraws', 'tvidraws', 'vdraws', 'sidraws', 'sijdraws', 'tsidraws'):
          del to_save[cat]
     configfile = Path(fname)
     with configfile.open("w") as tfile:
          for cat in to_save:
               tfile.write(str(cat)+": \n")
               tfile.write(str(to_save[cat])+"\n")
     # print(to_save)

File: RScripts/Walmart Example/summarize_output.R
summarize_fitp <- function(fitp){
  print(head(fitp$mdraws[1,])); print(mean(fitp$mdraws))
  print(head(fitp$sdraws[1,])); print(mean(fitp$sdraws))
  # print(head(fitp$mmean)); print(mean(fitp$mmean)) # Will be the same as the mean values I took down above
  # print(head(fitp$smean)); print(mean(fitp$smean))
  print(head(fitp$msd)); print(mean(fitp$msd))
  print(head(fitp$ssd)); print(mean(fitp$ssd))
  print(mean(fitp$m.5)); print(mean(fitp$m.lower)); print(mean(fitp$m.upper))
  print(mean(fitp$s.5)); print(mean(fitp$s.lower)); print(mean(fitp$s.upper))
}
summarize_fitv <- function(fitv){
  print(fitv$vdraws[1:60, ]); print(mean(fitv$vdraws))
  print(fitv$vdrawsh[1:60, ]); print(mean(fitv$vdrawsh))
  print(fitv$mvdraws); print(fitv$mvdrawsh)
  print(fitv$vdraws.sd); print(fitv$vdrawsh.sd); print(fitv$vdraws.5)
  print(fitv$vdrawsh.5); print(fitv$vdraws.lower); print(fitv$vdraws.upper)
  print(fitv$vdrawsh.lower); print(fitv$vdrawsh.upper)
}
summarize_fits <- function(fits){
  print(mean(fits$vidraws)); print(mean(fits$vijdraws))
  print(mean(fits$tvidraws)); print(mean(fits$vdraws))
  print(mean(fits$sidraws)); print(mean(fits$sijdraws))
  print(mean(fits$tsidraws))
  # print(fits$msi); print(fits$msi.sd); print(fits$si.5)
  # print(fits$si.lower); print(fits$si.upper); print(fits$msij)
  # print(fits$sij.sd); print(fits$sij.5); print(fits$sij.lower)
  # print(fits$sij.upper); print(fits$mtsi); print(fits$tsi.sd)
  # print(fits$tsi.5); print(fits$tsi.lower); print(fits$tsi.upper)
}

save_fit_obj <- function(fit, fname, objtype){
  if (objtype != 'fitp'){
    to_save <- data.frame(unlist(fit))
  }
  else {
    drop <- c("mdraws", "sdraws")
    temp <- fit[!(names(fit) %in% drop)]
    to_save <- data.frame(unlist(temp))
  }
  write.table(to_save, file = paste(fname, '.txt', sep=""), sep = "")
 
  if (objtype == 'fitp'){ # (Nothing needed for objtype == 'fit')
    to_save2 <- list()
    to_save2[1] <- paste("Mean(mdraws): ", toString(mean(fit$mdraws)), "\n", sep="")
    to_save2[2] <- paste("Mean(sdraws): ", toString(mean(fit$sdraws)), "\n", sep="")
    to_save2[3] <- paste("Mean(msd): ", toString(mean(fit$msd)), "\n", sep="")
    to_save2[4] <- paste("Mean(ssd): ", toString(mean(fit$ssd)), "\n", sep="")
    to_save2[5] <- paste("Mean(m_lower): ", toString(mean(fit$m.lower)), "\n", sep="")
    to_save2[4] <- paste("Mean(m_5): ", toString(mean(fit$m.5)), "\n", sep="")
    to_save2[6] <- paste("Mean(m_upper): ", toString(mean(fit$m.upper)), "\n", sep="")
    to_save2[7] <- paste("Mean(s_lower): ", toString(mean(fit$s.lower)), "\n", sep="")
    to_save2[8] <- paste("Mean(s_5): ", toString(mean(fit$s.5)), "\n", sep="")
    to_save2[9] <- paste("Mean(s_upper): ", toString(mean(fit$s.upper)), "\n", sep="")
    write.table(to_save2, file = paste(fname, '_summary.txt', sep = ""), sep="")
  }  else if (objtype == 'fits'){
    to_save2 <- list()
    to_save2[1] <- paste("Mean(vidraws): ", toString(mean(fit$vidraws)), "\n", sep="")
    to_save2[2] <- paste("Mean(vijdraws): ", toString(mean(fit$vijdraws)), "\n", sep="")
    to_save2[3] <- paste("Mean(tvidraws): ", toString(mean(fit$tvidraws)), "\n", sep="")
    to_save2[4] <- paste("Mean(vdraws): ", toString(mean(fit$vdraws)), "\n", sep="")
    to_save2[5] <- paste("Mean(sidraws): ", toString(mean(fit$sidraws)), "\n", sep="")
    to_save2[6] <- paste("Mean(sijdraws): ", toString(mean(fit$sijdraws)), "\n", sep="")
    to_save2[7] <- paste("Mean(tsidraws): ", toString(mean(fit$tsidraws)), "\n", sep="")
    to_save2[8] <- paste("Mean(msij): ", toString(mean(fit$msij)), "\n", sep="")
    to_save2[9] <- paste("Mean(sij_sd): ", toString(mean(fit$sij.sd)), "\n", sep="")
    to_save2[10] <- paste("Mean(sij_5): ", toString(mean(fit$sij.5)), "\n", sep="")
    to_save2[11] <- paste("Mean(sij_lower): ", toString(mean(fit$sij.lower)), "\n", sep="")
    to_save2[12] <- paste("Mean(sij_upper): ", toString(mean(fit$sij.upper)), "\n", sep="")
    write.table(to_save2, file = paste(fname, '_summary.txt', sep = ""), sep="")
  }
}




save_fits_old <- function(fits, fname){
  fits2 <- fits # A copy
  # Manual removal (was giving a lot of problems in the for loop):
  fits2$vidraws <- NULL; fits2$vijdraws <- NULL; fits2$tvidraws <- NULL
  fits2$vdraws <- NULL; fits2$sidraws <- NULL; fits2$sijdraws <- NULL
  fits2$tsidraws <- NULL
  # str(fits2)
  to_save <- data.frame(unlist(fits2))
  # str(to_save)
  write.table(to_save, file = fname, sep = "")
}